android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -176,10 +179,46 @@ class MusicService : HeadlessJsTaskService() {
                 )
 
                 player.notificationManager.createNotification(notificationConfig)
+
+                // publish progress update events if its configured
+                attemptProgressUpdateEvent(options)
+            }
+        }
+    }
+
+    private fun attemptProgressUpdateEvent(options: Bundle) {
+        cancelProgressUpdateRunnable()
+        val updateInterval = Utils.getIntOrNull(options, PROGRESS_UPDATE_EVENT_INTERVAL_KEY)
+        if (updateInterval != null && updateInterval > 0) {
+            val updateIntervalMs = (updateInterval * 1000).toLong()
+            progressUpdateRunnable = Runnable {
```

Comment:
Exoplayer does not have a progress updated callback, so usually using a Runner is the right approach but we're using Kotlin coroutines in this file so we'll want to us that instead. You can create a "timer" using flow.

Reply:
@dcvz updated this to a flow, let me know if it looks correct.